Code/Framework/AzCore/AzCore/Settings/SettingRegisteryUtility.h

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/Settings/SettingsRegistry.h>
+
+namespace AZ::SettingsRegistryUtils
+{
+    template<typename T>
+    inline void SetRegistry(SettingsRegistryInterface* registry, const AZStd::string_view setting, T&& value)
+    {
+        if (registry)
+        {
+            registry->Set(setting, AZStd::forward<T>(value));
+        }
+    }
+
+    template<typename T>
+    inline AZStd::remove_cvref_t<T> GetRegistry(SettingsRegistryInterface* registry, const AZStd::string_view setting, T&& defaultValue)
```

Comment:
question: Is there a reason why we're passing the `SettingsRegistryInterface` in? Can we not just call it internally (as it's an interface it's possible to swap out in tests if necessary too) - will just make the call sites a bit cleaner

Reply:
Yes actually.

The Settings Registry is not a singleton. There is a global instance that is available that can be queried from using `SettingsRegistry::Get()`, but local instances of the Settings Registry do exist and are being used.
For example the SettingsRegistryBuilder uses a local Settings Registry to isolate Settings that are applicable for the GameLauncher.

Another example is the [SettingsRegistryMergeUtils::VisitManifestJson](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/Settings/SettingsRegistryMergeUtils.cpp#L1451-L1468) function uses a local Settings Registry to load in the manifest.json files like `o3de_manifest.json`, `engine.json`, `project.json` and `gem.json` to read the manifest object names (ex. `project_name` from the project.json, `engine_name` from the engine.json, `gem_name` from the gem.json).